### PR TITLE
feat: configure RabbitMQ status action

### PR DESCRIPTION
**Pull request by martin-trajanovski**
## Description

- This PR aims to add the RabbitMQ status action with some simple configuration.
- It also removes the executed flag on the status actions because it is not that useful for now. We need a better way of handling the status action logs and preventing double execution on the same proposal if already executed.

## Motivation and Context

We want to be more explicit with sending messages to the RabbitMQ and being able to select triggers for sending messages to the exchanges.

## How Has This Been Tested

- e2e tests
- manual tests

## Fixes

https://jira.esss.lu.se/browse/SWAP-3529

## Changes

https://github.com/UserOfficeProject/user-office-core/assets/6333063/be5676f2-7c9a-4d90-995d-90f17b62cff6

## Tests included/Docs Updated?

- [x] I have added tests to cover my changes.
- [ ] All relevant doc has been updated

### CODE REVIEW

**Review by janosbabik — CHANGES_REQUESTED** [review]
Hi Martin! Just some small changes that I think would be good to have.

File: apps/backend/src/datasources/postgres/ProposalDataSource.ts [hunk 1]
```diff
@@ -819,64 +817,13 @@ export default class PostgresProposalDataSource implements ProposalDataSource {
       }
     }
 
-    if (shouldResetStatusActions) {
```
[the remark this answers is not in the document]
Comment (martin-trajanovski) [reply apps/backend/src/datasources/postgres/ProposalDataSource.ts:822]: This is removed so I think there is no need for transaction anymore I guess.

[the remark this answers is not in the document]
Comment (janosbabik) [reply apps/backend/src/datasources/postgres/ProposalDataSource.ts:822]: It is recommended to use a transaction when there are multiple queries that depend on each other. We want these queries (like a single unit) to run simultaneously to ensure data consistency and efficient handling by the database. In this scenario, there is a possibility of data changing between the time we retrieve the "call" or "status_changing_events" and when we update the "proposal_events".

**Review by janosbabik — APPROVED** [review]

**Comment by Scott-James-Hurley** [comment]
Is there a better way of exposing (or communicating) the ability to send a status action to a queue to a user? I don't think a User Officer should have to know what RabbitMQ is or what exchange is being used.

**Comment by Scott-James-Hurley** [comment]
> Is there a better way of exposing (or communicating) the ability to send a status action to a queue to a user? I don't think a User Officer should have to know what RabbitMQ is or what an exchange is.
> 
> image

Maybe the user should be just be told what the outcome will be of whatever consumes the event. This would be the description has to be configurable, maybe determined by the environment.

**Review by Scott-James-Hurley — APPROVED** [review]

**Comment by martin-trajanovski** [comment]
> > Is there a better way of exposing (or communicating) the ability to send a status action to a queue to a user? I don't think a User Officer should have to know what RabbitMQ is or what an exchange is.
> > image
> 
> Maybe the user should be just be told what the outcome will be of whatever consumes the event. This would be the description has to be configurable, maybe determined by the environment.

@Scott-James-Hurley I will open another ticket for this and we can discuss both within our team and on the planning meeting as well for improving the communication with the regular users so they can understand what this action outcome is and what to expect. Thanks for the feedback.

**Pull request merged**